src/dune_tui/dune_tui.ml

```diff
@@ -98,13 +158,62 @@ module Message_viewer = struct
         else I.(cropped_image </> I.char A.empty ' ' img_width 1)
   ;;
 
+  let open_editor loc =
+    match
+      let open Option.O in
+      let* loc = loc in
+      let env = Env.initial in
+      let* editor = Env.get env "EDITOR" in
+      let+ prog = Bin.which ~path:(Env_path.path env) editor >>| Path.to_string in
+      let argv =
+        let { Lexing.pos_fname = file_to_open; pos_lnum = line; pos_cnum = col; _ } =
+          Loc.start loc
+        in
+        (* Editor specific logic for opening files at a location. *)
+        (* TODO: should we pass [prog] or [editor] to argv.(0)? *)
+        match editor with
+        | "vi" | "vim" | "nvim" ->
+          let position_arg = String.concat ~sep:"" [ "+"; Int.to_string line ] in
+          [ prog; position_arg; file_to_open ]
+        | "emacs" ->
+          let position_arg =
+            String.concat ~sep:"" [ "+"; Int.to_string line; ":"; Int.to_string col ]
+          in
+          [ prog; position_arg; file_to_open ]
+        | "code" ->
+          let position_arg =
+            String.concat ~sep:":" [ file_to_open; Int.to_string line; Int.to_string col ]
+          in
+          [ prog; "-g"; position_arg ]
+        (* TODO: subl, nano, ... ? *)
+        | _ -> [ prog; file_to_open ]
+      in
+      prog, argv
+    with
+    | Some (prog, argv) ->
+      Terminal_manager.disable ();
+      let () =
+        let _pid, exit_status =
+          Dune_spawn.Spawn.spawn ~prog ~argv () |> Unix.waitpid []
```

Comment:
I think this is going to break the scheduler since it's raising against this with `waitpid`. You need to tell the scheduler to ignore this pid.